packages/react-sdk/src/hooks/useOptimisticDeviceStatus.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+interface Toggleable {
+  toggle: () => Promise<void>;
+}
+
+export function useOptimisticDeviceStatus(
+  isMute: boolean,
+  manager: Toggleable,
+) {
+  const [optimisticIsMute, setOptimisticIsMute] = useState(isMute);
+  const requestPromise = useRef<Promise<void> | null>(null);
+
+  useEffect(() => {
+    console.log({ optimisticIsMute, isMute });
+    if (optimisticIsMute !== isMute && requestPromise.current === null) {
+      requestPromise.current = manager.toggle().finally(() => {
+        requestPromise.current = null;
+      });
+    }
+  }, [manager, optimisticIsMute, isMute]);
```

Comment:
"Eventual consistency" :)

Update `optimisticIsMute` first, then deal with doing actual work in an effect.